lib/widgets/actions.dart

```diff
@@ -67,6 +67,24 @@ Future<void> markNarrowAsRead(BuildContext context, Narrow narrow) async {
   }
 }
 
+Future<void> markNarrowAsUnreadFromMessage(
+  BuildContext context,
+  Message message,
+  Narrow narrow,
+) async {
+  final zulipLocalizations = ZulipLocalizations.of(context);
+  await updateMessageFlagsStartingFromAnchor(
+    context: context,
+    startingAnchor: NumericAnchor(message.id),
```

Comment:
nit: Let's order the arguments consistently. `startingAnchor` as a parameter comes after `apiNarrow`; `onFailedTitle` comes after `progressMessage`.